### PR TITLE
mon: allow overriding the initial mon_host

### PR DESCRIPTION
**Pull request by batrick**
This overrides what the CephContext believes to be the current quorum of
monitors (retrieved from other instances of the MonClient), introduced
by [1]. Tests need to be able to target a specific monitor for
exercising forwarding and other things.

[1] 731e2db9fb4611f767446a3c8e778a097ce70d35
Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/common/options.cc [hunk 1]
```diff
@@ -429,6 +429,12 @@ std::vector<Option> get_global_options() {
     .set_flag(Option::FLAG_STARTUP)
     .add_service("common"),
 
+    Option("mon_host_override", Option::TYPE_STR, Option::LEVEL_DEV)
```
Comment (neha-ojha) [inline src/common/options.cc:432]: we should probably add release notes/docs about this.

**Review by neha-ojha — APPROVED** [review]
looks fine to me

File: src/common/options.cc (hunk 1 above)
Comment (batrick) [reply src/common/options.cc:432]: It's really a testing config option, I don't think we'd ever have a reason for users to config this?

Comment (neha-ojha) [reply src/common/options.cc:432]: I seem to recall there was a tracker issue from some user where this was helpful, but should only be needed for debugging purposes.

**Review by ShyamsundarR — COMMENTED** [review]
Thanks, LGTM!

File: src/common/options.cc (hunk 1 above)
Comment (batrick) [reply src/common/options.cc:432]: If you think it's possible for users to want it, I'll bump it to LEVEL_ADVANCED and add a release note.

Comment (batrick) [reply src/common/options.cc:432]: done!

**Comment by batrick** [comment]
jenkins render docs

**Comment by batrick** [comment]
jenkins render docs

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2020-09-18_17:47:08-fs-wip-pdonnell-testing-20200918.022453-distro-basic-smithi/

failures unrelated. Will run the relevant rados suite test next.

**Comment by batrick** [comment]
https://pulpito.ceph.com/pdonnell-2020-09-19_00:25:56-rados:standalone-wip-pdonnell-testing-20200918.022453-distro-basic-smithi/

**Pull request merged**